Dockerfile.must-gather

```diff
@@ -1,4 +1,4 @@
-FROM quay.io/openshift/origin-must-gather:4.16.0 AS builder
+FROM quay.io/openshift/origin-must-gather:4.17.0 AS builder
```

Comment:
should be 4.18 if the image is available (it usually is not when we update)

Reply:
it is:
https://quay.io/repository/openshift/origin-must-gather?tab=tags
updated it. Thanks!